Sources/X509/OCSP/OCSPPolicy.swift

```diff
@@ -31,8 +31,51 @@ public protocol OCSPRequester: Sendable {
     /// - Parameters:
     ///   - request: DER-encoded request bytes
     ///   - uri: uri of the OCSP responder
-    /// - Returns: DER-encoded response bytes
-    func query(request: [UInt8], uri: String) async throws -> [UInt8]
+    /// - Returns: DER-encoded response bytes if they request was successful or a terminal or non-terminal error.
+    func query(request: [UInt8], uri: String) async -> OCSPRequesterQueryResult
+}
+
+public struct OCSPRequesterQueryResult: Sendable {
+    @usableFromInline
+    enum Storage: Sendable {
+        case success([UInt8])
+        case nonTerminal(Error)
+        case terminal(Error)
+    }
+    @usableFromInline
+    var storage: Storage
+    
+    @inlinable
+    init(_ storage: Storage) {
+        self.storage = storage
+    }
+}
+
+extension OCSPRequesterQueryResult {
+    /// The OCSP query is considered successful and has returned the given DER-encoded response bytes.
+    /// - Parameter bytes: DER-encoded response bytes
+    @inlinable
+    public static func response(_ bytes: [UInt8]) -> Self {
```

Comment:
Hm, how do you actually get the bytes out of the result with the new API? Similar question on the other APIs.

Reply:
Its just meant to be consumed internally by `OCSPVerificationPolicy`

Reply:
Ah so nobody is going to call `query` but rather we are calling it for them. So they only need to be able to construct the return type but not handle it.

Reply:
exactly!